src/planner/space_constraint.c

```diff
@@ -117,11 +117,11 @@ is_valid_scalar_space_constraint(ScalarArrayOpExpr *op, List *rtable)
 }
 
 static FuncExpr *
-make_partfunc_call(PartitioningFunc partfunc, List *args, Oid inputcollid)
+make_partfunc_call(PartitioningFunc *partfunc, List *args, Oid inputcollid)
```

Comment:
Since the intention is to only read the data, it is good to encode this in the signature.
```suggestion
make_partfunc_call(const PartitioningFunc *partfunc, List *args, Oid inputcollid)
```